src/components/invite/Invite.js

```diff
@@ -135,11 +134,7 @@ const ShareBox = ({ level }) => {
   const shareUrl = `${Config.publicUrl}?inviteCode=${inviteCode}`
   const bounty = result(level, 'bounty.toNumber', 100) / 100
 
-  const share = useMemo(
-    () =>
-      isSharingAvailable ? generateShareObject(shareTitle, shareMessage, shareUrl) : `${shareMessage}\n${shareUrl}`,
-    [shareUrl],
-  )
+  const share = useMemo(() => generateShareObject(shareTitle, shareMessage, shareUrl), [shareUrl])
```

Comment:
why do we're always generating an object ?
the new sharing flow is the following:

```javascript

import { useClipboardCopy } from '../../lib/hooks/useClipboard'
import useNativeSharing from '../../lib/hooks/useNativeSharing'

import {
  generateReceiveShareObject,
  generateReceiveShareText,
  generateShareLink,
  isSharingAvailable,
} from '../../lib/share'

// ...

const shareOrText = useMemo(() => {
    const factory = isSharingAvailable ? generateReceiveShareObject : generateReceiveShareText

    return factory(code, amount, counterPartyDisplayName, fullName)
  }, [code, amount, counterPartyDisplayName, fullName])

  const shareHandler = useNativeSharing(shareOrText)
  const copyHandler = useClipboardCopy(shareOrText)

// ...
          <ShareLinkReceiveAnimationButton
            onPress={isSharingAvailable ? shareHandler : copyHandler}
          />
```

Reply:
the animation is specific and not suitable for this design
the logic for transforming the object to text is in the sharebutton